### PR TITLE
Add liberate test and build_test invoke tasks (bug 1365907)

### PR DESCRIPTION
**Pull request by zalun**
Links are created to allow updating the phutil_map file.
As file volumes in docker require to be provided with an absolute path,
/tmp/phutil_map is used. It needs to be linked to phutil_map to update
it directly from `arc liberate` command.

For the `arc unit` tests to run there is a need to have database
running. The phabdb from conduit-demo was used to create a minimum
viable container.

### CODE REVIEW

File: tasks.py [hunk 1]
```diff
@@ -42,3 +42,26 @@ def imageid(ctx):
...
     ))
+
+def link_phutil_map(ctx):
+    ctx.run("ln -fs `pwd`/phutil_map /tmp/phutil_map")
+
+@task
+def build_test(ctx):
+    """Test phabricator extensions."""
+    link_phutil_map(ctx)
+    ctx.run("docker-compose -f docker-compose.test.yml build phabricator")
```
Comment (ckolos) [inline tasks.py]: I can't find any mention of circle supporting docker-compose in the 1.0 product. I only see support for it in the 2.0 beta.

Comment (ckolos) [inline tasks.py]: ah, belay that; while they don't explicitly cover it in the docs, the Ubuntu 14.04 image does include compose 1.5.2. Ignore me.

**Comment by ckolos** [comment]
looks reasonable to me!

**Comment by zalun** [comment]
Forgotten to add the `docker-compose.test.yml` file.

**Comment by darkwing** [comment]
I was initially getting the following error:

```
$ invoke liberate
Cannot start service phabricator: oci runtime error: container_linux.go:247: starting container process caused "process_linux.go:359: container init caused \"rootfs_linux.go:54: mounting \\\"/tmp/phutil_map\\\" to rootfs \\\"/mnt/sda1/var/lib/docker/aufs/mnt/2e40325e8fff73ceb2b6604911989a6801110d893bbc03c9d6ffc3571f368c26\\\" at \\\"/mnt/sda1/var/lib/docker/aufs/mnt/2e40325e8fff73ceb2b6604911989a6801110d893bbc03c9d6ffc3571f368c26/app/phabricator/src/__phutil_library_map__.php\\\" caused \\\"not a directory\\\"\""
: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type
dwalsh-29224:phabricator-extensions dwalsh$ sudo rm -rf /tmp/phutil_map
```

So I switched `/tmp/` to my local user home directory and got the following output:

```
$ invoke test
+ cd phabricator
+ test -n phabdb
+ /app/wait-for-mysql.php
Set 'mysql.port' in local configuration.
Set 'mysql.user' in local configuration.
+ set +x
+ test -n test-ext
README
auth
conduit
differential
+ /app/arcanist/bin/arc unit extensions/auth/__tests__/PhabricatorBMOAuthAdapterTestCase.php extensions/auth/__tests__/PhabricatorBMOAuthProviderTestCase.php extensions/differential/customfield/__tests__/DifferentialBugzillaIdCustomFieldTestCase.php
   PASS    2ms★  PhabricatorBMOAuthAdapterTestCase::testBugzillaNameParserFindsNick
   PASS    2ms★  PhabricatorBMOAuthProviderTestCase::testThrowExceptionShowsGenericErrorWhenDebugIsOff
   PASS   2.2s   PhabricatorBMOAuthProviderTestCase::testUniqueAuthTokenGenerated
   PASS   <1ms★  DifferentialBugzillaBugIDCustomFieldTestCase::testBugIdPropertyViewRendering
```

I'm assuming it's just my machine having the issue.  R+

**Comment by mars-f** [comment]
I think we can land this, but I also have a suggestion for an alternate approach to the `phutil_map` file linking problem.  We could place the phutil_map file in a project subdirectory, like `phabextmap/phutil_map`.  That relative directory can then be linked into the container at `/usr/lib/phabricator/extmap/` or something.  Finally, an entrypoint.sh script would symlink the `/usr/lib/` path to `/app/phabricator/src/__phutil_library_map__.php`.

The advantage would be we don't have to rely on pyinvoke machinery as much to run the project.  It would also break the dependency on having a `/tmp` directory on the host.

**Pull request merged**